distribution/lib/Standard/Base/0.0.0-dev/src/System/File_By_Line.enso

```diff
@@ -0,0 +1,221 @@
+        read_line self line
+
+    ## Reads the first line
+    first : Text
+    first self = self.get 0
+
+    ## Reads the first line
+    second : Text
+    second self = self.get 1
+
+    ## Counts the number of lines in the file.
+    count : Integer
+    count self =
+        end_at = if self.limit_lines.is_nothing then -1 else self.limit_lines
+        for_each_lines self 0 end_at Nothing
+        ## We've added all the indexes to the row map including the last one so we need to subtract 1
+           As row_map can be shared if we have a limit return that.
+        if end_at == -1 then self.row_map.getSize-1 else end_at.min self.row_map.getSize-1
+
+    ## Returns the lines in the file as a vector.
+    to_vector : Vector Text
+    to_vector self = File_Error.handle_java_exceptions self.file <|
+        end_at = if self.limit_lines.is_nothing then Nothing else self.limit_lines-1
+        FileLineReader.readLines self.java_file self.file_end self.row_map 0 end_at self.charset self.filter_func
+
+    ## Performs an action on each line.
+
+       Arguments
+       - function: The action to perform on each line.
+    each : (Text -> Any) -> Nothing
+    each self function =
+        new_function _ t = function t
+        self.each_with_index new_function
+
+    ## Performs an action on each line.
+
+       Arguments
+       - function: The action to perform on each line.
+    each_with_index : (Integer -> Text -> Any) -> Nothing
+    each_with_index self function =
+        end_at = if self.limit_lines.is_nothing then Nothing else self.limit_lines-1
+        for_each_lines self 0 end_at function
+
+    ## Transforms each line in the file and returns the result as a vector.
+
+       Arguments
+       - action: The action to perform on each line.
+    map : (Text -> Any) -> Vector Any
+    map self action =
+        builder = Vector.new_builder
+        wrapped_action _ t = builder.append (action t)
+        self.each_with_index wrapped_action
+        builder.to_vector
+
+    ## Transforms each line in the file and returns the result as a vector.
+
+       Arguments
+       - action: The action to perform on each line.
+    map_with_index : (Integer -> Text -> Any) -> Vector Any
+    map_with_index self action =
+        builder = Vector.new_builder
+        wrapped_action i t = builder.append (action i t)
+        self.each_with_index wrapped_action
+        builder.to_vector
+
+    ## Skips the specified number of lines.
+
+       Arguments
+       - lines: The number of lines to skip.
+    skip : Integer -> File_By_Line
+    skip self lines:Integer =
+        ## Read the line
+        create_row_map parent line =
+            position = if parent.row_map.getSize > line then parent.row_map.get line else
+                for_each_lines self 0 line Nothing
+                parent.row_map.get parent.row_map.getSize-1
```

Comment:
minor nitpick feel free to ignore
but if I understand correctly, can this be simplified to:
```suggestion
            if parent.row_map.getSize <= line then for_each_lines self 0 line Nothing
            position = parent.row_map.get line
```

Reply:
Not quite but can do something similar.